plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -479,6 +500,8 @@ fu_algoltek_usb_device_write_firmware(FuDevice *device,
 	if (!fu_algoltek_usb_device_rst(self, 0x100, error))
 		return FALSE;
 
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
```

Comment:
Does this mean that you can call and activate callback?

Reply:
After the updating, the device would reactivate itself automatically. No need to unplug and replug the device. The device would disconnect itself then reconnect to the computer.
I add this flag to avoid showing "device version not updated on success..." after updating.

Reply:
The purpose for this flag isn't to avoid that message - it's to indicate update is a two step process.

To avoid that message, can you re-read firmware version when device is done updating?

Reply:
The device need to reactivate to get the updated firmware version, but this warning message will display first.

Does there any flag that you suggest to add or could I add some delay to postpone the version check?

Reply:
What does reactivate mean? Does it disappear (disconnect from host) and come back? Or just need to wait to read version again?

Reply:
The second one. The device will disappear and reconnect.

Reply:
You should use FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG then

Reply:
Thanks! Updated.